### ISSUE TITLE
[java] RedundantFieldInitializer FP with expressions that start with zero

### ISSUE THREAD

**Issue by oowekyala** [pmd/pmd#3743]
**Affects PMD Version:** 6.41.0

**Rule:** [RedundantFieldInitializer](https://pmd.github.io/latest/pmd_rules_java_performance.html#redundantfieldinitializer)

**Description:**

See #3684:

> As @dykov pointed out, 0 + 1 - 1 is only flagged by chance. Expressions starting with 0 but not evaluating to zero (eg 0+1) would cause a false positive which should be fixed in pmd 6.

**Code Sample demonstrating the issue:**

```java
class Impl {
    int b = 0 + 1; // FP
}
```

**Comment by dykov** [comment]
@oowekyala, wouldn't it be more useful to forbid such operations in field initialization?
I mean, if you have AdditiveExpression of only Literals?

**Comment by oowekyala** [comment]
I think this is not the purpose of RedundantFieldInitializer and forbidding simple expressions should be done by another rule (if it should be done at all). I don't think it's very useful to have a rule for that, when in many cases constant arithmetic expressions are easier to read than a literal.

**Comment by dykov** [comment]
Yes, by another rule, sure.
I think such operation make sence only with String literals to concat them.

**Comment by jsotuyod** [comment]
This is no longer reproducible under PMD 7.
